### ISSUE TITLE
v1.8.3 - Dynamic import that can't execute causes error when using import maps.

### ISSUE THREAD

**Issue by JoakimCh** [denoland/deno#10168]
```js
if (never) {
  await import('whatever')
}
```
I get the `unmapped bare specifier` error. But this should not cause any error or else it will be impossible to write code which supports several runtimes.

Basically I want to check if it's running in Node.js and then to load a Node.js specific module ONLY in that case. When running in Deno or the browser then I don't need to do this, hence Deno should not complain about the code in the block which is never executed.

**Comment by JoakimCh** [comment]
I'll follow up with a solution for my case where I wanted to import Node,js's "crypto". Adding it like this makes the error go away. But still I do NOT approve of the error...
```json
{
  "imports": {
     "crypto": "node:crypto"
  }
}
```

**Comment by kitsonk** [comment]
We should defer import map errors for dynamic imports, like we defer other errors until the import is actually imported.

**Comment by nayeemrmn** [comment]
Aside from undesired static analysis and the proposed fix in #10374, it seems like a smell that this manifests as an "import map error" at all. Unmapped bare specifiers should be let through as they are and hit an error in the normal resolver. That would give a nice error with a location.
